### ISSUE TITLE
query --output build prints None labels as null

### ISSUE THREAD

**Issue by benjaminp** [bazelbuild/bazel#12589]
```
$ touch WORKSPACE
$ cat > BUILD
alias(
   name = "t",
   actual = select({":setting": ":a", "//conditions:default": None}),
)

config_setting(
   name = "setting",
   define_values = {"a": "1"},
)
$ bazel query --output build //:t
alias(
  name = "t",
  actual = select({"//:setting": "//:a", "//conditions:default": null}),
)
```

**Comment by gregestren** [comment]
Rationale for me changing the label: `team-Configurability` owns `cquery` issues while @meisterT via `team-Performance` owns `query` ones.

I realize there's overlap which sometimes muddles this. For example, both commands support `--output=build` and `select` is clearly `cquery-`-related. But I think this particular report could only happen through `query` (unless `None` appears outside of a `select` somehow).

**Comment by benjaminp** [comment]
This is still a bug.

**Comment by meisterT** [comment]
I had a quick look today, this is coming from here: https://cs.opensource.google/bazel/bazel/+/master:src/main/java/net/starlark/java/eval/Printer.java;l=165;drc=675172439552055398c480990e277bb2f39d6aaa

(last touched by @brandjon in f107a53c916d4f6f1fab0fa011ebf9c1d920b1c7)

**Comment by fmeum** [comment]
@meisterT I looked into this as well and I think that the map returned by `Selector#mapCopy` at https://cs.opensource.google/bazel/bazel/+/master:src/main/java/com/google/devtools/build/lib/query2/query/output/BuildOutputFormatter.java;l=214 is incorrect: For select cases that map to `None`, the returned value is the default of the attribute (which is `null` for the Java-defined `alias` rule), not the value specified by the user in the `BUILD` file, due to ffa14a6a372b22b16a9ac12f3d78a1b2859043cc. 

This leads me to think that this is actually a combination of two issues: `BuildOutputFormatter` not formatting `null` values returned by `Selector#mapCopy` correctly and `Selector#mapCopy` not setting values to `null` rather than their defaults.

**Comment by brandjon** [comment]
Thanks Tobi and Fabian for the diagnosis.

@gregestren It's concerning to me that select dicts can now contain null. Are there other places where a select may be inspected within Starlark? Two cases come to mind: 1) any API we have or may add to allow selects to be manipulated within macros; 2) any API we use to expose raw attrs containing selects to starlark-defined rule transitions (though I think we previously agreed we wouldn't do that, and would instead if anything only allow them access to configured results of select()s, if it wouldn't cause a cycle).

Is there any logical distinction between `null` and `None` in any attribute type? For instance, if you set a `label` attribute to `None`, does it appear "explicitly-specified" to native code (i.e. `isAttributeValueExplicitlySpecified`)? If so, we have a problem in that this semantic difference should be observable to anyone inspecting the select, so "null" appears to be a valid thing to see even if you can't write it that way literally in a BUILD file. If not, then can we store `None` instead of `null` internally in the select?

Of course we can patch the custom `Printer` in `BuildOutputFormatter.outputRawAttrValue`, but that's only papering over the problem if the above isn't resolved.

**Comment by gregestren** [comment]
Right, I don't believe 2) is possible, or planned to be possible. 

https://github.com/bazelbuild/bazel/commit/ffa14a6a372b22b16a9ac12f3d78a1b2859043cc documents that `None` means `isAttributeValueExplicitlySpecified == false`.

**Comment by benjaminp** [comment]
still alive

**Comment by fmeum** [comment]
@benjaminp May have been fixed by 38ade70fed4b34043d21a606717ad2ed1ee9c696

**Comment by benjaminp** [comment]
Unfortunately, that commit does not alter this bug.
